### ISSUE TITLE
Weird SQLite Error on Startup

### ISSUE THREAD

**Issue by adamkrellenstein** [CounterpartyXCP/counterparty-core#1655]
```
13|counter | [2024-04-14 08:28:05][INFO] Running v10.1.0-rc.1 of counterparty-lib.
13|counter | [2024-04-14 08:28:05][INFO] Running v10.1.0-rc.1 of counterparty-server.
13|counter | [2024-04-14 08:28:05][INFO] Connecting to address indexer.
13|counter | [2024-04-14 08:28:05][INFO] Acquiring lock.
13|counter | [2024-04-14 08:28:05][INFO] Connecting to database (SQLite 3.45.1.0).
13|counter | ERROR:root:SQLITE_LOG: recovered 642 frames from WAL file /home/adam/.local/share/counterparty/counterparty.db-wal (283) SQLITE_NOTICE_RECOVER_WAL
13|counter | [2024-04-14 08:28:05][INFO] Starting API Server.
13|counter | [2024-04-14 08:28:05][INFO] Resuming parsing.
13|counter | [2024-04-14 08:28:06][INFO] New block inserted 839147

13|counterparty  | ERROR:root:SQLITE_LOG: statement aborts at 22: [
13|counterparty  |         SELECT * FROM balances
13|counterparty  |         WHERE (address = ? AND asset = ?)
13|counterparty  |         ORDER BY rowid DESC LIMIT 1
13|counterparty  |     ] database schema has changed (17) SQLITE_SCHEMA
13|counterparty  | [2024-04-14 08:28:21][INFO] Block 839147 parsed, ledger hash is 8a8ee58ad33a530a54630cc57c251ff9f7eb22827867c016ae67bd79b592bfc6 and txlist hash is 9c664aa0a9e17d3b569fac54ddc42809dc2c338a62d12a07352d7953e88b9909
13|counterparty  | [2024-04-14 08:28:21][INFO] Block: 839147 (15.67, hashes: L:2bfc6 / TX:b9909 / M:8ce3b)
13|counterparty  | [2024-04-14 08:28:24][INFO] New block inserted 839148
13|counterparty  | [2024-04-14 08:28:27][ERROR] Key error in addrindexrs still exists!!!!!
13|counterparty  | [2024-04-14 08:28:27][WARNING] tx missing in rawtx cache: '1cd237a14fb463ff2ff8fae0dd62126b2bd1d1d140c94f1486c26d880fa40b49' -- txhash_list size: 2990, hash: b2bc4ded94244c52f4caa994baf374e5 / raw_transactions_cache size: 2990 / # rpc_batch calls: 2990 / txhash in noncached_txhashes: True / txhash in txhash_list: True -- list []
13|counterparty  | [2024-04-14 08:28:32][INFO] Credit 1 A3552076589448406000 to 171EqT3RUgyLbwRDhmMfhXPnLws4AXLWxy
```

**Comment by mattmarcello** [comment]
What I think is going on:

- DB didnt shut down correctly, SQLite is doing what is should and recovering from write ahead log 
- DB connection is opened in `read-only` mode and as such doesn't need to acquire writer lock ( which could not be acquired until after recovery is completed ) 
- Query is made against table which is being modified by WAL recovery process, error thrown 

Thoughts @ouziel-slama?

**Comment by ouziel-slama** [comment]
hum.. I don't think so.. I fixed the bug which caused the presence of a wal file and the error message appears even when there is no recovery.

**Comment by mattmarcello** [comment]
Hmm, I think we should change the `"Lock acquired."` log to `info` level.  
```
# server.py#50
def get_lock():
    logger.info("Acquiring lock.")

    # Cross‐platform.
    if platform.system() == "Darwin":  # Windows or OS X
        # Not database‐specific.
        socket_family = socket.AF_INET
        socket_address = ("localhost", 8999)
        error = "Another copy of server is currently running."
    else:
        socket_family = socket.AF_UNIX
        socket_address = "\0" + config.DATABASE
        error = (
            f"Another copy of server is currently writing to database {config.DATABASE}"
        )

    lock_socket = socket.socket(socket_family, socket.SOCK_DGRAM)
    try:
        lock_socket.bind(socket_address)
    except socket.error:
        raise LockingError(error)  # noqa: B904
    logger.debug("Lock acquired.")
```

**Comment by ouziel-slama** [comment]
(That said, it's been a while since I saw this error message!)

**Comment by mattmarcello** [comment]
Even though we don't have a callstack here, it's something like this:

```
ledger.get_balance
<message>.parse 
blocks.parse_tx
blocks.parse_block
blocks.follow
server.start_all
```

This would all happen before DB recovers from WAL file.

**Comment by ouziel-slama** [comment]
<img width="744" alt="Capture d’écran 2024-04-18 à 17 42 51" src="https://github.com/CounterpartyXCP/counterparty-core/assets/4574979/58585754-d6a8-44c9-859a-449c63efb3a6">
this happens systematically on the first RPC call, but not on the next one! and there is no WAL recovery.

**Comment by ouziel-slama** [comment]
```
ERROR:root:SQLITE_LOG: statement aborts at 43: [
        SELECT d.*, a.asset_longname
        FROM dispensers d
        LEFT JOIN assets a ON a.asset_name = d.asset
        WHERE (tx_hash = ?)
        ORDER BY d.rowid DESC LIMIT 1
  (17) SQLITE_SCHEMA
```
FWIW I get this error when the first call is `get_dispenser_info`!

**Comment by mattmarcello** [comment]
Solved by  #1701
